ui/src/source.ts

```diff
@@ -140,24 +166,56 @@ export const commits = commitsStore.readable;
 const objectStore = remote.createStore<SourceObject>();
 export const object = objectStore.readable;
 
-const revisionsStore = remote.createStore<Revisions>();
+// Updated when the project store changes (i.e. when the user navigates to a new project)
+const revisionsStore = remote.createStore<PeerRevisions[]>();
 export const revisions = revisionsStore.readable;
+revisions.subscribe(store => {
+  if (store.status === remote.Status.Success) {
+    // The first `PeerRevisions` in the response belongs to the default peer.
+    const defaultRevisions = store.data[0];
+
+    const peerId = defaultRevisions.identity.peerId;
+    updateCurrentPeerId({ peerId });
+
+    // Now that we have a peer, set the current revision to the default branch.
+    // If not found, use the first branch returned from proxy.
+    const defaultRevision =
+      defaultRevisions.branches.find(
+        branch => branch.name === defaultRevisions.defaultBranch
+      ) || defaultRevisions.branches[0];
+    updateCurrentRevision({ revision: defaultRevision });
+  }
+});
 
+// Local stores
 export const objectType = writable(ObjectType.Tree);
 export const resetObjectType = (): void => objectType.set(ObjectType.Tree);
 export const objectPath = writable(null);
 export const resetObjectPath = (): void => objectPath.set(null);
-export const currentRevision = writable(null);
-export const resetCurrentRevision = (): void => currentRevision.set(null);
-export const currentPeerId = writable(null);
-export const resetCurrentPeerId = (): void => currentPeerId.set(null);
+
+export const currentRevision = writable<Branch | Tag | undefined>(undefined);
```

Comment:
```suggestion
export const currentRevision = writable<SupportedRevision | undefined>(undefined);
```

We could re-use the above `type` right?